mocha-casperjs.js

```diff
@@ -47,8 +47,13 @@ module.exports = function (Mocha, casper, utils) {
     })
   })
 
+  function clearSteps() {
+    casper.step = 0
+    casper.steps = []
+  }
+
   casper.on('waitFor.timeout', function(timeout, details) {
-    casper.step++
```

Comment:
Test should be aborted if it fails. This line should do the trick, but  it is not enough. If test fails on some sub step (level > 1), current test will skip current step and will continue after it.